spark-jobs/src/main/scala/za/co/absa/enceladus/conformance/HyperConformance.scala

```diff
@@ -57,8 +58,8 @@ class HyperConformance (implicit cmd: ConfCmdConfig,
     val conformance = dao.getDataset(cmd.datasetName, cmd.datasetVersion)
 
     val conformedDf = DynamicInterpreter.interpret(conformance, rawDf)
-      .withColumnIfDoesNotExist(InfoDateColumn, infoDateColumn)
-      .withColumnIfDoesNotExist(InfoDateColumnString, date_format(infoDateColumn,"yyyy-MM-dd"))
+      .withColumnIfDoesNotExist(InfoDateColumn, coalesce(infoDateColumn, current_timestamp().cast(DateType)))
+      .withColumnIfDoesNotExist(InfoDateColumnString, coalesce(date_format(infoDateColumn,"yyyy-MM-dd"), lit("")))
```

Comment:
This `coalesce` is needed to make created fields non-nullable. This is so that we keep compatibility with the publish Avro schema that we used when the report date was just a constant.
I've put `coalesce` here and removed from the code above so that 'enceladus_info_date' and `enceladus_info_date_string` are coalesced separately.

Reply:
Interesting. Maybe would worth explanation in the code.